### PR TITLE
Improve io parallelism

### PR DESCRIPTION
**Pull request by polarker**

### CODE REVIEW

File: flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala [hunk 1]
```diff
@@ -73,14 +83,60 @@ abstract class ChainHandler[T <: FlowData: Serde, S <: InvalidStatus, Command](
...
+        }
+      case Left(error) => handleIOError(error)
+    }
+  }
+
+  def addDataToBlockFlow(data: T): IOResult[AVector[TransactionTemplate]]
+
+  def broadcastReadyTxs(txs: AVector[TransactionTemplate]): Unit = {
+    if (txs.nonEmpty) {
+      // TODO: maybe broadcast it based on peer sync status
```
Comment (tdroxler) [inline flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala:106]: should we have a ticket for it?

**Review by tdroxler — APPROVED** [review]

**Comment by tdroxler** [comment]
LGTM, I'm just wondering how hard would it be to have a rough estimation of the progression of the sync? At least to have the info of the real total of blocks when we start syncing, so we can make the math ourself :)

File: flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala (hunk 1 above)
Comment (polarker) [reply flow/src/main/scala/org/alephium/flow/handler/ChainHandler.scala:106]: There is already one in the pool

**Comment by polarker** [comment]
> LGTM, I'm just wondering how hard would it be to have a rough estimation of the progression of the sync? At least to have the info of the real total of blocks when we start syncing, so we can make the math ourself :)

The main problem is you cannot trust the number from peers

**Pull request merged**